update genesis

Garandor opened the pull request:
## Description

relates to OR closes: #926 

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] This PR is targeted against the *current*  Milestone ( otherwise discuss if it can be added in time)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.


Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

ferrell-code commented on node/src/chain_specs/manta.rs:
```diff
@@ -157,30 +173,38 @@ fn manta_dev_genesis(
-        aura_ext: Default::default(),
         parachain_system: Default::default(),
-        polkadot_xcm: manta_runtime::PolkadotXcmConfig {
+        polkadot_xcm: calamari_runtime::PolkadotXcmConfig {
```
manta_runtime?

Garandor replied:
thanks, fixed!

ferrell-code commented on node/src/chain_specs/manta.rs:
```diff
@@ -198,4 +222,198 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+                    .collect(),
+            },
+            parachain_system: Default::default(),
+            polkadot_xcm: calamari_runtime::PolkadotXcmConfig {
```
manta_runtime?

Garandor replied:
thanks, fixed!

Garandor commented on node/src/chain_specs/manta.rs:
```diff
@@ -198,4 +222,198 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+                INITIAL_CONTROLLER_BALANCE,
+            ), // COLLATOR 1
+            (
+                hex!("0202d1760e33f60f977fcff88b611cb95e66d0b3b74c12ac6c8037b8b9764a31").into(),
```
@grenade pls doublecheck the collator addresses match with what you gave me

grenade replied:
all verified (c1 - c5, node, collator, nimbus, vrf).

grenade replied:
all verified (c1 - c5, node, collator, nimbus, vrf).

Garandor commented on node/src/chain_specs/manta.rs:
```diff
@@ -198,4 +222,198 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+
+    pub fn manta_mainnet_config() -> MantaChainSpec {
+        let genesis_collators: Vec<Collator> = vec![
+        Collator::new( // c1: dfWh1oFWNxrmrHdiGdVpQQD6Pp3SmY5NrNsAGNRCxgaqJuZif
```
@grenade pls doublecheck that collator addresses, session keys and nodeID match with what you gave me

grenade replied:
all verified (c1 - c5, node, collator, nimbus, vrf).

grenade replied:
all verified (c1 - c5, node, collator, nimbus, vrf).

ferrell-code commented on node/src/chain_specs/manta.rs:
```diff
@@ -198,4 +222,203 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
 /// Returns the Manta mainnet chainspec.
 pub fn manta_config() -> Result<MantaChainSpec, String> {
     MantaChainSpec::from_json_bytes(&include_bytes!("../../../genesis/manta-genesis.json")[..])
+    // Ok(mainnet::manta_mainnet_config())
```
Remove commented line?

Garandor replied:
this is a bit of a topic for discussion, i want to preserve the code that generated the genesis json.
If possible, i'd actually want to keep 225 active and delete 224.
tbh I don't even know the reason why we do this file read in the first place.

@ghzlatarev do you know the history?

I can only imagine that the `mod mainnet` code will eventually break compilation if we don't keep it updated, so this file preserves the record of mainnet as it was used to spawn the network even when we have to comment out/remove the code that generates this file

ghzlatarev replied:
When we make it into a file we can use it with any version of the node and it should start from the original runtime.
If it's in-place it would start with the latest runtime version.
I think that's the reason.
Also maybe obfuscating some account ids.

Garandor replied:
makes sense, the RT blob gets built into the file.

This actually means we can only generate the config after bumping spec versions, otherwise it'll report as `manta/4000`

I'm wondering how best to preserve this code tho, maybe splitting the module into its own file?
I don't think we need to obfuscate any of these pubkeys

ferrell-code replied:
I think its because the genesis file must be etched into stone can never be changed once network is live. 

Aka if we change the genesis than the node can't sync to manta network as it has a different genesis block

I think leaving this comment here is good thing, as we may want to change the genesis before launching manta network

Garandor replied:
the line will be needed in #957 and we can use that issue to figure out where to put the code that generated it.

Commit:
Pared down manta runtime genesis

- Parachain ID 2100
- production keys
- move `from_seed_unchecked` from primitives into runtime, Manta RT has 2 keys, Calamari/Dolphin have AuraId as extra key
- remove service_aura since manta RT no longer uses it

Signed-off-by: Adam Reif <Garandor@manta.network>

grenade reviewed (approved):
all verified (c1 - c5, node, collator, nimbus, vrf).

Commit:
Merge branch 'manta' into garandor/manta-genesis

zqhxuyuan commented on node/src/chain_specs/manta.rs:
```diff
@@ -221,4 +222,203 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+        let genesis_collators_clone = genesis_collators.clone(); // so we can move it into the constructor closure
+
+        MantaChainSpec::from_genesis(
+            "Shark Parachain",
```
rename to manta parachain? or the manta symbol is Shark?

zqhxuyuan commented on node/src/chain_specs/manta.rs:
```diff
@@ -221,4 +222,203 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+            ), // COLLATOR 5
+            (
+                root_key.clone(),
+                TOTAL_ISSUANCE - 4 * INITIAL_CONTROLLER_BALANCE - 4 * INITIAL_COLLATOR_BALANCE,
```
is this calculation right? we have 3 controller, and 5 collator? or 4 controller and 4 collator?

zqhxuyuan commented on node/src/chain_specs/manta.rs:
```diff
@@ -221,4 +222,203 @@ pub fn manta_testnet_config() -> Result<MantaChainSpec, String> {
+            (
+                hex!("0619e4018531059b677d70b476b72431d3e88fe71d2c27bba5d945975e824956").into(),
+                INITIAL_CONTROLLER_BALANCE,
+            ), // COLLATOR 1
```
should be CONTROLLER4. and bellow also need to rename the right order

Garandor replied:
good point, the name `INITIAL_CONTROLLER_BALANCE` is misleading here.
We just decided to give one collator more starting balance for emergency recovery.
It is not an actual multisig contoller account, i'd keep it this way, hope this explanation is enough
